### PR TITLE
Prefab instancing flags

### PR DESCRIPTION
**Pull request by Erlangist**
Introduced new parameter to PrefabReference::SetPrefab(): instanceFlags, which allows the instantiated prefab to appear in game the same as it appears in Editor. instanceFlags controls, which attributes of the root prefab node is copied to the scene node, associated with prefab instance.

### CODE REVIEW

File: Source/Urho3D/Scene/PrefabReference.h [hunk 1]
```diff
@@ -37,6 +37,20 @@ enum class PrefabInlineFlag
...
+/// By default, no copying is made.
+enum class PrefabInstanceFlag
+{
+    None = 0,
+    SetNodeScale = 1 << 0,
+    SetNodePosition = 1 << 1,
+    SetNodeRotation = 1 << 2,
+    SetNodeTags = 1 << 3,
+    SetNodeName = 1 << 4,
+    SetNodeVariables = 1 << 5,
```
Comment (gleblebedev) [inline Source/Urho3D/Scene/PrefabReference.h:50]: would be nice to have flag "All" here

File: Source/Urho3D/Scene/PrefabReference.cpp [hunk 2]
```diff
@@ -230,15 +230,50 @@ void PrefabReference::RemoveInstance()
...
 }
 
-void PrefabReference::InstantiatePrefab(const NodePrefab& nodePrefab)
+void PrefabReference::InstantiatePrefab(const NodePrefab& nodePrefab, PrefabInstanceFlags instanceFlags)
 {
     const auto flags = PrefabLoadFlag::KeepExistingComponents | PrefabLoadFlag::KeepExistingChildren
         | PrefabLoadFlag::LoadAsTemporary | PrefabLoadFlag::IgnoreRootAttributes;
     PrefabReaderFromMemory reader{nodePrefab};
     node_->Load(reader, flags);
+    if (instanceFlags != PrefabInstanceFlag::None) {
```
Comment (gleblebedev) [inline Source/Urho3D/Scene/PrefabReference.cpp:239]: Is all of this necessary? Can we just have 1 flag to copy parameters or not to copy? Is there a use case where users would prefer to set only subset of the attributes?

Comment (Erlangist) [reply Source/Urho3D/Scene/PrefabReference.cpp:239]: There may be such cases, it's better to have control over what's copied and what's not.
And I think that it's better to write explicitly what you want to copy than spending time in debugger trying to figure out why that tag/rotation/... appeared in the node.

**Review by eugeneko — APPROVED** [review]
I made it a bit shorter, looks okay

**Review by eugeneko — APPROVED** [review]

**Pull request merged**